### PR TITLE
HBW-84 Use higher order component instead of HBWSelectMixin

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

File: hbw/app/javascript/packs/hbw/components/form/select_table.js.jsx [hunk 1]
```diff
@@ -1,154 +1,174 @@
...
-        if (this.isEqual(id, value)) {
-          selected = 'selected';
+      buildCssFromConfig(config) {
+        return `text-align-${config.alignment}`;
+      },
+
+      getChoices () {
+        let choices;
+
+        if (this.props.params.mode === 'select') {
```
Comment (mxsmirnoff) [inline hbw/app/javascript/packs/hbw/components/form/select_table.js.jsx]: Does select_table have "lookup" type? 
I moved this from mixin here and removed extra option "lookup", as I think.

**Comment by mxsmirnoff** [comment]
Updated

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 2]
```diff
@@ -232,8 +232,20 @@ modulejs.define('HBWFormSelect',
...
+      },
+
+      getChoices (value) {
+        if (this.props.params.mode === 'select') {
+          this.addCurrentValueToChoices(value);
+
+          return this.props.params.choices.slice();
+        } else if (this.props.params.mode === 'lookup') {
+          return this.props.params.choices;
+        }
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]:
```
else {
  return null;
}
```

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 3]
```diff
@@ -232,8 +232,20 @@ modulejs.define('HBWFormSelect',
...
+
+        return null;
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: You don't need this line.

File: hbw/app/javascript/packs/hbw/components/form/select_table.js.jsx [hunk 4]
```diff
@@ -151,10 +151,22 @@ modulejs.define('HBWFormSelectTable',
...
+          choices = this.props.params.choices.slice();
+
+          if (this.props.params.nullable) {
+            this.addNullChoice(choices);
+          }
+
+          return choices;
+        }
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form/select_table.js.jsx]:
```
} else {
  return null;
}
```

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx (hunk 2 above)
Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: It is case like this, isn't it?
```
// bad
function foo() {
  if (x) {
    return x;
  } else {
    return y;
  }
}

// good
function foo() {
  if (x) {
    return x;
  }

  return y;
}

```
https://github.com/airbnb/javascript#blocks--no-else-return

Comment (TimAle) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: It is, but it is not our style :)
https://github.com/latera/homs/blob/5e4fba2a0497145959cfa34117fc9f86e4e2b03e/.eslintrc#L25

Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: But it's not mentioned in https://github.com/latera/homs/blob/master/CONTRIBUTING.md

**Comment by mxsmirnoff** [comment]
Updated

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx (hunk 2 above)
Comment (TimAle) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Ok: https://github.com/latera/homs/pull/208

Comment (mxsmirnoff) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Nice

Comment (sugarfree1) [reply hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Thanks all, that's awesome now.

File: hbw/app/javascript/packs/hbw/components/helpers/select.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+export default WrappedComponent => React.createClass({
+  displayName: `withSelect(${getDisplayName(WrappedComponent)})`,
+
+  getChosenValue () {
+    if (this.props.params.mode === 'select') {
+      if (this.props.value === null) {
+        if (this.props.params.nullable) {
+          return null;
+        } if (this.props.params.choices.length) {
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/helpers/select.js]: ```else if``` maybe?

File: hbw/app/javascript/packs/hbw/components/helpers/select.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          if (Array.isArray(first)) {
+            return first[0];
+          }
+          return first;
+        }
+        return null;
+      }
+      return this.props.value;
+    }
+    return this.props.value;
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/helpers/select.js]:
```
else {
  return this.props.value;
}
```

File: hbw/app/javascript/packs/hbw/components/helpers/select.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    return a.toString() === b.toString();
+  },
+
+  isChoiceEqual (choice, value) {
+    if (Array.isArray(choice)) {
+      return this.isEqual(choice[0], value);
+    }
+    return this.isEqual(choice, value);
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/helpers/select.js]: else

**Comment by mxsmirnoff** [comment]
Updated

**Comment by mxsmirnoff** [comment]
Updated

**Pull request merged**